barretenberg/cpp/src/barretenberg/honk/proof_system/permutation_library.hpp

```diff
@@ -166,4 +166,229 @@ void compute_permutation_grand_products(std::shared_ptr<typename Flavor::Proving
     });
 }
 
+/**
+ * @brief Compute new polynomials which are the concatenated versions of other polynomials
+ *
+ * @details Multilinear PCS allow to provide openings for concatenated polynomials in an easy way by combining
+ * commitments. This method creates concatenated version of polynomials we won't need to commit to. Used in Goblin
+ * Translator
+ *
+ * @tparam Flavor
+ * @tparam StorageHandle
+ * @param proving_key Can be a proving_key or an AllEntities object
+ */
+template <typename Flavor, typename StorageHandle> void compute_concatenated_polynomials(StorageHandle* proving_key)
+{
+    // Concatenation groups are vectors of polynomials that are concatenated together
+    auto concatenation_groups = proving_key->get_concatenation_groups();
```

Comment:
can we replace `auto` with a proper type? It would make it easier to read what is happening here. If the type is complex, a `using` declaration would be better than `auto` as the reader would know where to look to read the type definition.

Reply:
Done